chain-core/src/property.rs

```diff
@@ -35,6 +35,21 @@
 //! is selected to write a block in the chain.
 //!
 
+use std::hash::Hash;
+
+/// Trait identifying the block identifier type.
+pub trait BlockId: Eq + Hash {}
+
+/// Trait identifying the block date type.
+pub trait BlockDate: Eq + Ord {}
+
+/// Trait identifying the transaction identifier type.
+pub trait TxId: Eq + Hash {}
+
+/// Trait identifying the block header type.
+/// TODO: provide header in the data model.
+pub trait Header {}
```

Comment:
no we cannot provide this. the Header type is very specific to a given implementation of the blockchain. This is only needed for your implementation of the network. Can we keep it in there (if you want with a Todo reminding us that this may need to be generalised).

Reply:
This is needed for the `GetHeaders` gRPC method. Should we drop it altogether, or should it be optionally supported, depending perhaps on a dynamic property of `network_core::Node`?